Javier/base login component front and back

ProwlingLynx opened the pull request:
Migrated image dir into public to squash outstanding bug. Added login component logic for sending data and receiving responses. Added back end logic to accommodate new login logic. Edited route from GET /login to Post /login. Noted a bug in database query sending positive response for incorrect password submission. Noted an error from socket.io instrument if npm install is not run in root. Added error logic in POST /login. Corrected data handling for POST /login. 

Note: State is not updating to reflect login behavior. This is intended as another ticket. The scope of this work is to create the base behavior for user login auth system and database integration in data flow.

Note: Multiple PR's at the time of this PR. Likely suspect this branch to be stale at the time of review.

silvered-cat committed:
Implemented some logic and local states to login modal.

silvered-cat committed:
Added state to login component. Moved image dir into public dir and adjusted timer component to match new dir. debugged login component.

silvered-cat committed:
implemented socket server to http server post route. debugged modal closing. added login logic closing modal in login component.

silvered-cat committed:
Refreshing stale branch

Merge branch 'main' of https://github.com/Shady-Wolves/Sus-Puppies into modalWerk

silvered-cat committed:
Resolved merge conflict.

Dareitus reviewed (changes requested):
Conflicts due to Stale Branch.

Please pull origin and compare your changes to stale changes before resubmitting

silvered-cat committed:
Refreshing feature branch and resolved conflicts in GamePage.jsx and Login.jsx

Merge branch 'main' of https://github.com/Shady-Wolves/Sus-Puppies into modalWerk

ghost commented:
I pulled down the new changes from main and resolved the conflicts from GamePage and Login components.

mwolf2531 commented:
Looks good now
